### PR TITLE
test: Shifted admin check function and added integration test

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Part of #5320

#### Short description of what this resolves:
This moves the admin rights function to the helpers and adds a test to check it

#### Changes proposed in this pull request:

- Shifted helper function
- Added integration test

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/api/helpers/auth.py [hunk 1]
```diff
@@ -31,3 +31,14 @@ def is_verified_user():
...
         return current_user.is_authenticated
+
+    @staticmethod
+    def check_auth_admin(username, password):
+        # This function is called to check for proper authentication & admin rights
+        if username and password:
+            user = User.query.filter_by(_email=username).first()
+            if user:
+                if user.is_correct_password(password):
+                    if user.is_admin:
```
Comment (poush) [inline app/api/helpers/auth.py]: Hi @mrsaicharan1, I think its better that lines 40, 41, 41 be reduced to only 1

File: tests/all/integration/api/helpers/test_auth.py [hunk 2]
```diff
@@ -45,5 +45,17 @@ def test_is_accessible(self):
             logout_user()
             self.assertEqual(auth_manager.is_accessible(), False)
 
+    def test_check_auth_admin(self):
```
Comment (poush) [inline tests/all/integration/api/helpers/test_auth.py]: I think you should also test with admin=False

**Comment by mrsaicharan1** [comment]
@poush Will do that :)

File: tests/all/integration/api/helpers/test_auth.py (hunk 2 above)
Comment (poush) [reply tests/all/integration/api/helpers/test_auth.py]: You can create two users and one admin and other non-admin and this should give correct response for each of them

File: app/api/helpers/auth.py (hunk 1 above)
Comment (shreyanshdwivedi) [reply app/api/helpers/auth.py]: @mrsaicharan1 yes please use `and` and combine the logics

**Comment by mrsaicharan1** [comment]
Yeah two test cases. Agreed

File: tests/all/integration/api/helpers/test_auth.py (hunk 2 above)
Comment (mrsaicharan1) [reply tests/all/integration/api/helpers/test_auth.py]: @poush Made the requested changes. Please have a look

File: tests/all/integration/api/helpers/test_auth.py [hunk 3]
```diff
@@ -45,5 +45,22 @@ def test_is_accessible(self):
...
+        """Method to test proper authentication & admin rights for a user"""
+
+        with app.test_request_context():
+            auth_manager = auth.AuthManager()
+            auth_manager.init_login(app)
+            user = create_user(email='authtest@gmail.com', password='password')
+            user.is_admin = True
+            status = auth_manager.check_auth_admin('authtest@gmail.com', 'password')
+            self.assertEqual(status, True)
```
Comment (iamareebjamal) [inline tests/all/integration/api/helpers/test_auth.py]:
```suggestion
            self.assertEqual(True, status)
```

File: tests/all/integration/api/helpers/test_auth.py [hunk 4]
```diff
@@ -45,5 +45,22 @@ def test_is_accessible(self):
...
+
+            user = create_user(email='authtest2@gmail.com', password='password')
+            user.is_admin = False
+            status = auth_manager.check_auth_admin('authtest2@gmail.com', 'password')
+            self.assertEqual(status, False)
```
Comment (iamareebjamal) [inline tests/all/integration/api/helpers/test_auth.py]:
```suggestion
            self.assertEqual(False, status)
```

**Comment by mrsaicharan1** [comment]
@shreyanshdwivedi @uds5501 @poush @iamareebjamal Made the suggested changes. Please have a look now.

**Pull request merged**